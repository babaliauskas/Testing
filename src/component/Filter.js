import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pizza from './Pizza';

class Filter extends Component {
  constructor(props) {
    super(props);

    this.state = {
      filter: '',
      sort: true
    };

    this.handleFilter = this.handleFilter.bind(this);
  }

  handleFilter(event) {
    this.setState({ filter: event.target.value });
  }

  render() {
    let displayPizzas;
    let filteredPizzas = [];
    let filterSortBtn = null;
    const { filter } = this.state;
    const { loading, pizzas, handleSortPizzas } = this.props;

    // Checking if pizzas loaded

    if (pizzas.length > 0) {
      filterSortBtn = (
        <div>
          <input
            type="text"
            onChange={event => this.handleFilter(event)}
            value={filter}
          />

          <button onClick={handleSortPizzas}>Sort Pizzas</button>
        </div>
      );
    }

    // Filtering pizzas and storing to a new variable
    if (pizzas) {
      pizzas.forEach(pizza => {
        if (pizza.toLowerCase().includes(filter.toLowerCase())) {
          return filteredPizzas.push(pizza);
        }
      });
    }

    // Displaying List of pizzas if it is already loaded
    if (loading) {
      displayPizzas = <p>Loading...</p>;
    } else {
      displayPizzas = filteredPizzas.map((pizza, index) => (
        <Pizza key={index} pizza={pizza} />
      ));
    }

    return (
      <div>
        {filterSortBtn}
        <ul>{displayPizzas}</ul>
      </div>
    );
  }
}

Filter.propTypes = {
  loading: PropTypes.bool.isRequired,
  pizzas: PropTypes.array.isRequired,
  handleSortPizzas: PropTypes.func.isRequired
};

export default Filter;
