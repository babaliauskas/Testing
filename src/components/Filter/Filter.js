import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pizza from '../Pizza/Pizza';
import FilterInput from './FilterInput/FilterInput';

class Filter extends Component {
  constructor(props) {
    super(props);

    this.state = {
      filter: '',
    };

    this.handleFilter = this.handleFilter.bind(this);
  }

  handleFilter(event) {
    this.setState({ filter: event.target.value });
  }

  render() {
    const filteredPizzas = [];
    const { filter } = this.state;
    const { pizzas, handleSortPizzas } = this.props;

    // Filtering pizzas and storing to a new variable
    if (pizzas) {
      pizzas.forEach((pizza) => {
        if (pizza.toLowerCase().includes(filter.toLowerCase())) {
          filteredPizzas.push(pizza);
        }
      });
    }

    return (
      <div>
        {pizzas.length > 0 ? (
          <FilterInput
            handleFilter={this.handleFilter}
            pizzas={pizzas}
            filter={filter}
            handleSortPizzas={handleSortPizzas}
          />
        ) : null}
        <ul>
          <Pizza filteredPizzas={filteredPizzas} />
        </ul>
      </div>
    );
  }
}

Filter.propTypes = {
  pizzas: PropTypes.instanceOf(Array).isRequired,
  handleSortPizzas: PropTypes.func.isRequired,
};

export default Filter;
