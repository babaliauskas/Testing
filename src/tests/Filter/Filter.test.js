/* global it, describe, expect, jest, beforeEach */
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Filter from '../../components/Filter/Filter';

configure({ adapter: new Adapter() });

describe('Filter', () => {
  let filteredPizzas;
  const mockSortPizzas = jest.fn();
  const props = {
    pizzas: ['Sausage', 'Cheese'],
    handleSortPizzas: mockSortPizzas,
  };
  const filter = shallow(<Filter {...props} />);

  beforeEach(() => {
    filteredPizzas = props.pizzas;
  });

  it('should `filteredPizzas` equal to props.pizzas ', () => {
    expect(filteredPizzas).toBe(props.pizzas);
  });

  it('should render properly', () => {
    expect(filter).toMatchSnapshot();
  });

  it('should contain `Pizza` component', () => {
    expect(filter.find('Pizza').exists()).toBe(true);
  });

  it('should contain `FilterInput` component', () => {
    expect(filter.find('FilterInput').exists()).toBe(true);
  });

  describe('Checking proptypes', () => {
    it('Pizzas list should be array', () => {
      expect(Array.isArray(props.pizzas)).toBe(true);
    });

    it('handleSortPizzas should be function', () => {
      expect(typeof props.handleSortPizzas).toBe('function');
    });
  });
});
