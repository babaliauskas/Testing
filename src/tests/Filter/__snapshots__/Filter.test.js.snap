// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    handleSortPizzas={[MockFunction]}
    pizzas={
      Array [
        "Sausage",
        "Cheese",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilterInput
          filter=""
          handleFilter={[Function]}
          handleSortPizzas={[MockFunction]}
        />,
        <ul>
          <Pizza
            filteredPizzas={
              Array [
                "Sausage",
                "Cheese",
              ]
            }
          />
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "filter": "",
          "handleFilter": [Function],
          "handleSortPizzas": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Pizza
            filteredPizzas={
              Array [
                "Sausage",
                "Cheese",
              ]
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filteredPizzas": Array [
              "Sausage",
              "Cheese",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilterInput
            filter=""
            handleFilter={[Function]}
            handleSortPizzas={[MockFunction]}
          />,
          <ul>
            <Pizza
              filteredPizzas={
                Array [
                  "Sausage",
                  "Cheese",
                ]
              }
            />
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filter": "",
            "handleFilter": [Function],
            "handleSortPizzas": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Pizza
              filteredPizzas={
                Array [
                  "Sausage",
                  "Cheese",
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "filteredPizzas": Array [
                "Sausage",
                "Cheese",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
